File: scripts/generate_sitemap.py
import os

domain = "https://creatorhelpertool.vercel.app"
root_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))

html_files = []
for root, dirs, files in os.walk(root_dir):
    # Skip certain directories
    dirs[:] = [d for d in dirs if d not in ['.git', 'node_modules', 'images', 'scripts']]
    for file in files:
        if file.endswith('.html') and not file.startswith('google') and file != 'transcript.html':
            html_files.append(os.path.join(root, file))

sitemap_content = """<?xml version="1.0" encoding="UTF-8"?>
<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
"""

for file_path in html_files:
    rel_path = os.path.relpath(file_path, root_dir).replace('\\', '/')
    url_path = rel_path
    priority = "0.5"
    changefreq = "monthly"
    
    if url_path == "index.html":
        url_path = ""
        priority = "1.0"
        changefreq = "weekly"
    elif "about.html" in url_path:
        priority = "0.7"
        changefreq = "monthly"
    elif "contact.html" in url_path:
        priority = "0.6"
        changefreq = "monthly"
    elif "privacy.html" in url_path or "terms.html" in url_path:
        priority = "0.3"
        changefreq = "yearly"
        
    sitemap_content += f"""  <url>
    <loc>{domain}/{url_path}</loc>
    <changefreq>{changefreq}</changefreq>
    <priority>{priority}</priority>
  </url>
"""

sitemap_content += "</urlset>\n"

with open(os.path.join(root_dir, 'sitemap.xml'), 'w', encoding='utf-8') as f:
    f.write(sitemap_content)

print("sitemap.xml successfully generated!")

File: scripts/generate_sitemap.js
const fs = require('fs');
const path = require('path');

const DOMAIN = 'https://creatorhelpertool.vercel.app';

function getHtmlFiles(dir, fileList = []) {
  const files = fs.readdirSync(dir);
  files.forEach(file => {
    const filePath = path.join(dir, file);
    const stat = fs.statSync(filePath);
    if (stat.isDirectory()) {
      if (file !== 'node_modules' && file !== '.git' && file !== 'images' && file !== 'scripts') {
        getHtmlFiles(filePath, fileList);
      }
    } else if (file.endsWith('.html') && !file.startsWith('google') && file !== 'transcript.html') {
      fileList.push(filePath);
    }
  });
  return fileList;
}

const rootDir = path.resolve(__dirname, '..');
const htmlFiles = getHtmlFiles(rootDir);

let sitemapContent = `<?xml version="1.0" encoding="UTF-8"?>
<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
`;

htmlFiles.forEach(file => {
  const relativePath = path.relative(rootDir, file).replace(/\\/g, '/');
  let urlPath = relativePath;
  let priority = '0.5';
  let changefreq = 'monthly';

  if (urlPath === 'index.html') {
    urlPath = '';
    priority = '1.0';
    changefreq = 'weekly';
  } else if (urlPath.includes('about.html')) {
    priority = '0.7';
    changefreq = 'monthly';
  } else if (urlPath.includes('contact.html')) {
    priority = '0.6';
    changefreq = 'monthly';
  } else if (urlPath.includes('privacy.html') || urlPath.includes('terms.html')) {
    priority = '0.3';
    changefreq = 'yearly';
  }

  sitemapContent += `  <url>
    <loc>${DOMAIN}/${urlPath}</loc>
    <changefreq>${changefreq}</changefreq>
    <priority>${priority}</priority>
  </url>
`;
});

sitemapContent += `</urlset>\n`;

fs.writeFileSync(path.join(rootDir, 'sitemap.xml'), sitemapContent);
console.log('sitemap.xml successfully generated!');
